### ISSUE TITLE
async-listener compatibility

### ISSUE THREAD

**Issue by madbence** [request/request#1708]
```js
const cls = require('continuation-local-storage');
const req = require('request');
const http = require('http');
const sess = cls.createNamespace('foo');

sess.run(function () {
  sess.set('foo', 'bar');
  req('http://google.com', function () {
    console.log('req', sess.get('foo')); // prints undefined, but it should be bar
  });
  setTimeout(function () {
    console.log('timer', sess.get('foo')); // prints bar
  }, 100);
  http.get('http://google.com', function () {
    console.log('http', sess.get('foo')); // prints bar
  });
});
```

I'm not sure why did `request` lost the context :confused: 

```
$ node -pe 'process.versions'
{ http_parser: '2.5.0',
  node: '2.5.0',
  v8: '4.2.77.21',
  uv: '1.6.1',
  zlib: '1.2.8',
  ares: '1.10.1-DEV',
  modules: '44',
  openssl: '1.0.2d' }
```

**Comment by madbence** [comment]
Maybe related to othiym23/async-listener#42

**Comment by jkrems** [comment]
This *might* be fixed starting with `async-listener@0.5.6`. @madbence Can you confirm?

**Comment by madbence** [comment]
:+1: it works now!
